### DISCUSSION TITLE
ReferenceError: Supabase is not defined

### DISCUSSION THREAD

**Discussion by paulnovacovici** [supabase/supabase#32307] [Questions] [answered]
Getting Supabase undefined in my edge function after it's been working for months. Haven't pushed any changes to my application

```
/// <reference types="https://esm.sh/@supabase/functions-js/src/edge-runtime.d.ts" />

import { createClient } from "https://esm.sh/@supabase/supabase-js@2.7.1";
import { RecursiveCharacterTextSplitter } from "https://esm.sh/langchain@0.0.52/text_splitter";

import { corsHeaders } from "../_shared/cores.ts";
import { Database } from "../_shared/supabase.ts";
import { getDomain, isUserPremium } from "../_shared/utils.ts";

// @ts-expect-error TODO: Support Supabase AI types.
const model = new Supabase.ai.Session("gte-small");
```

**Comment by andy-cb-smith** [comment]
Same thing has happened to me.

**Reply by paulnovacovici** [reply]
Any workaround found?

**Reply by andy-cb-smith** [reply]
No, everything runs fine locally, guess something has been moved? I've sent a ticket.

**Comment by laktek** [comment, marked as answer]
@paulnovacovici @andy-cb-smith Sorry, this was due to a faulty release from our end. We have merged a fix (https://github.com/supabase/edge-runtime/pull/467) and will do a deploy within the next couple of hours.

**Reply by paulnovacovici** [reply]
Thanks @laktek! Is there anything I can do on my end to mitigate risk from crashing edge function due to faulty release in the future?

**Comment by laktek** [comment]
We've deployed the fix. Your functions should work again (no need to re-deploy them). 

For local dev/testing, please update the CLI to version 2.2.1 (`npx supabase@beta`)

**Comment by cpannwitz** [comment]
This problem still persists for us.
While being in VSCode, the types are found correctly. But using `deno lint && deno check`, the types are not found.

```
/// <reference types="jsr:@supabase/functions-js/edge-runtime.d.ts" />

import "jsr:@supabase/functions-js/edge-runtime.d.ts";

export function createSupabaseRuntimeAISession() {
  return new Supabase.ai.Session("gte-small");
}
```
